Lib/ast.py

```diff
@@ -557,6 +558,36 @@ def __new__(cls, *args, **kwargs):
 # We unparse those infinities to INFSTR.
 _INFSTR = "1e" + repr(sys.float_info.max_10_exp + 1)
 
+class _Precedence(IntEnum):
+    """Precedence table that originated from python grammar."""
+
+    TUPLE = auto()
+    YIELD = auto()           # 'yield', 'yield from'
+    TEST = auto()            # 'if'-'else', 'lambda'
+    OR = auto()              # 'or'
+    AND = auto()             # 'and'
+    NOT = auto()             # 'not'
+    CMP = auto()             # '<', '>', '==', '>=', '<=', '!=',
+                             # 'in', 'not in', 'is', 'is not'
+    EXPR = auto()
+    BOR = EXPR               # '|'
+    BXOR = auto()            # '^'
+    BAND = auto()            # '&'
+    SHIFT = auto()           # '<<', '>>'
+    ARITH = auto()           # '+', '-'
+    TERM = auto()            # '*', '@', '/', '%', '//'
+    FACTOR = auto()          # unary '+', '-', '~'
+    POWER = auto()           # '**'
+    AWAIT = auto()           # 'await'
+    ATOM = auto()
+
+    def next(self, by=1):
```

Comment:
You never use the `by` argument, so why have it?

Reply:
I was using it in here 
https://github.com/isidentical/cpython/blob/4576f36413ff5dd53b54674c68212cc3f95ff2be/Lib/ast.py#L1143-L1148
like this
```py
            left_precedence = operator_precedence.next(by=int(rassoc))
            right_precedence = operator_precedence.next(by=int(not rassoc))
```
but I believe current implementation is simpler. If you agree, I can remove that parameter and keep the current one

Reply:
> but I believe current implementation is simpler. If you agree, I can remove that parameter and keep the current one

I would suggest to remove it as is not clear to me what it does without checking the function (maybe a better name will do, but I think a function call `next` should not do much more than just give me the next value :)